sphinx/directives/other.py

```diff
@@ -39,6 +39,7 @@ class TocTree(Directive):
     final_argument_whitespace = False
     option_spec = {
         'maxdepth': int,
+        'name': str,
```

Comment:
That might work well. I'm mostly just worried about breaking backwards compat. I agree that it's not officially supported, but if we can avoid changing the semantics, it seems best.